packages/astro/src/@types/astro.ts

```diff
@@ -2054,6 +2055,101 @@ export interface AstroUserConfig {
 		 * For a complete overview, and to give feedback on this experimental API, see the [Rerouting RFC](https://github.com/withastro/roadmap/blob/feat/reroute/proposals/0047-rerouting.md).
 		 */
 		rewriting?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.env
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.10.0
+		 * @description
+		 *
+		 * Enables experimental `astro:env` features .
+		 *
+		 * The `astro:env` API lets you configure a type-safe schema for your environment variables, and indicate whether they should be available on the server or the client. Import and use your defined variables from the appropriate `/client` or `/server` module:
+		 *
+		 * ```js
+		 * ---
+		 * import { PUBLIC_POKEAPI } from "astro:env/client"
+		 * 
+		 * const data = fetch(`${PUBLIC_POKEAPI}/pokemon/squirtle`);
+		 * ---
+		 * To define the data type and properties of your environment variables, declare a schema in your Astro config in `experimental.env.schema`. The `envField` helper allows you define your variable as a string, number, or boolean and pass properties in an object:
+		 * 
+		 * ```js
+		 * // astro.config.mjs
+		 * import { defineConfig, envField } from "astro/config"
+		 * 
+		 * export default defineConfig({
+		 *     experimental: {
+		 *         env: {
+		 *             schema: {
+		 *                 PUBLIC_API_URL: envField.string({ context: "client", access: "public", optional: true }),
+		 *                 PUBLIC_PORT: envField.number({ context: "server", access: "public", default: 4321 }),
+		 *                 API_SECRET: envField.string({ context: "server", access: "secret" }),
+		 *             }
+		 *         }
+		 *     }
+		 * })
+		 * ```
+		 * 
+		 * There are currently 3 data types supported: strings, numbers and booleans.
+		 * 
+		 * There are three kinds of variables, determined by the combination of `context` (`client` or `server`) and `access` (`private` or `public`) settings defined in your [`env.schema`](#experimentalenvschema):
+		 * 
+		 * - **Public client variables**: These variables end up in both your final client and server bundles, and can be accessed from both client and server through the `astro:env/client` module:
+		 * 
+		 *     ```js
+		 *     import { PUBLIC_API_URL } from "astro:env/client"
+		 *     ```
+		 * 
+		 * - **Public server variables**: These variables end up in your final server bundle and can be accessed on the server through the `astro:env/server` module:
+		 * 
+		 *     ```js
+		 *     import { PUBLIC_PORT } from "astro:env/server"
+		 *     ```
+		 * 
+		 * - **Secret server variables**: These variables are not part of your final bundle and can be accessed on the server through the `astro:env/server` module:
+		 * 
+		 *     ```js
+		 *     import { getSecret } from "astro:env/server"
+		 * 
+		 *     const API_SECRET = getSecret("API_SECRET") // typed
+		 *     const SECRET_THAT_MAY_EXIST = getSecret("SECRET_THAT_MAY_EXIST") // string | undefined
```

Comment:
```suggestion
		 *     const SECRET_THAT_MAY_EXIST = getSecret("SECRET_THAT_MAY_EXIST") // string | undefined
```
I have to admit, I'm not clear what this does, or what "that may exist" means. Checking the RFC, it looks like this is what happens when something is not specified in the schema.

a) Are people going to be using `getSecret` for something NOT defined in their schema? If so when/why? What would this actually DO for people? I'm not sure of the use case for this.
b) Do we need to SHOW using `getSecret` for something not in their schema? Is this a case of "not everything that is true belongs in docs" or is this a very common thing people need to know how to do?

Reply:
Yeah the naming is probably bad, let me reexplain.

Basically in most cases, you'll want to define a server/secret var in your config schema for types, validation etc. But in some cases, it's also fine to get those purely dynamically. Maybe you allow a user to specify the key, maybe the key comes from a database...For all those cases, you'd want to use something like `process.env`.

But one of the point of Astro Env is to abstract that part so you can switch adapters without worrying. That's why you'll want to use `getSecret` with a key not defined in a schema

Reply:
So what about something like

```
const DYNAMIC_SECRET = getSecret("DYNAMIC_SECRET") // string | undefined
```
Is that sufficient enough to make sense?

```
const SECRET_NOT_IN_SCHEMA = getSecret("SECRET_NOT_IN_SCHEMA") // string | undefined
```
??

Anything along those lines?

Reply:
I like the 2nd suggestion more since a secret is "dynamic" 😅

Reply:
That looks good, ship it!